src/ops/fuelup_check.rs

```diff
@@ -56,34 +57,22 @@ fn format_version_comparison(current_version: &Version, latest_version: &Version
 }
 
 fn check_plugin(plugin_executable: &Path, plugin: &str, latest_version: &Version) -> Result<()> {
-    match std::process::Command::new(plugin_executable)
-        .arg("--version")
-        .output()
-    {
-        Ok(o) => {
-            let output = String::from_utf8_lossy(&o.stdout).into_owned();
-            match output.split_whitespace().last() {
-                Some(v) => {
-                    let version = Version::parse(v)?;
-                    info!(
-                        "{:>4}- {} - {}",
-                        "",
-                        bold(plugin),
-                        format_version_comparison(&version, latest_version)
-                    );
-                }
-                None => {
-                    info!("{:>4}- {} - Error getting version string", "", plugin);
-                }
-            };
+    match get_bin_version(plugin_executable) {
```

Comment:
nit: we can make it even shorter! This function never returns an error, so doesn't need to return an empty result.

```rust
fn check_plugin(plugin_executable: &Path, plugin: &str, latest_version: &Version) {
    let version_or_err = match get_bin_version(plugin_executable) {
        Ok(version) => format_version_comparison(&version, latest_version),
        Err(err) => err.to_string()
    };
    info!("{:>4}- {} - {}", "", plugin, version_or_err);
}
```